docs/experimental-features.md

```diff
@@ -0,0 +1,285 @@
+
+## Strategies to add new APIs for experimental features
+
+There are three main **recommended** strategies to implement APIs for new
+features:
+
+- If the feature requires a field on the API, or the feature may affect several
+  parts of the API, modify the stable APIs adding the necessary fields. These
+  fields **must** be optional, and it should be clearly stated in the godoc that
+  the feature is experimental and may change. In the CRD yaml the field(s)
+  should not be specified, and `x-kubernetes-preserve-unknown-fields` should be
+  used instead. Then, in the webhook, you can reject resources with fields
+  related to experimental features when validating the input CR using
+  [experimental.ValidateAPIFields()](https://github.com/knative/eventing/blob/917f57f0e68abbd0256c5fa4bff1ce2e83ffe78b/pkg/apis/experimental/api_validation.go#L13)
+- Alternative to the above approach, if the feature affects a whole resource,
+  and the API is a single value, that is not an object or an array of values,
+  use an annotation. Then, in the webhook, you can reject resources with
+  annotations related to experimental features when validating the input CR
+  using
+  [experimental.ValidateAnnotations()](https://github.com/knative/eventing/blob/917f57f0e68abbd0256c5fa4bff1ce2e83ffe78b/pkg/apis/experimental/api_validation.go#L38).
+  This approach is not suggested in case adding an API field is doable as
+  described above, but there might be some situations when this is not possible,
+  where an annotation is more suited as the feature API. For example:
+  [Eventing Kafka Broker - Configuring the order of delivered events](https://github.com/knative/docs/blob/main/docs/eventing/broker/kafka-broker.md#configuring-the-order-of-delivered-events)
+- If a feature is supposed to be opt-in even after its graduation to stable, for
+  example an optional behavior for a specific component/set of components, make
+  sure to have a flag to enable it in the place wherever is more appropriate for
+  that flag to exist. E.g. a flag to enable the broker to send an additional
+  header, when dispatching events, should be placed in the ConfigMap that
+  configures the broker (`Broker.Spec.Config`). This makes sure the experimental
+  flags are not confused with opt-in/opt-out stable features and fits into the
+  already existing/implemented scoping of components configuration. The
+  experimental flag to enable such a feature acts as a kill-switch, while the
+  flag in the appropriate ConfigMap acts as the effective "feature interface".
+
+It’s not required to follow such strategies, but we encourage feature designers
+to stick to one of these approaches, wherever possible, in order to ensure
+consistency and aligned user experience.
+
+## Feature stages
+
+### Stage definition
+
+A feature can be in Alpha, Beta or GA stage. An Alpha feature means:
+
+- Disabled by default.
+- No vendor is required to implement the new feature at this point, but it’s
+  encouraged to.
```

Comment:
All depends on whether the feature is normative or not. A vendor is never required to implement a feature, unless the feature is normative and the vendor wants to be conformant wrt the new feature. 

Maybe drop this line and say by definition an experimental feature is not normative so vendors don't have to support it in order to be compliant.

Reply:
In this commit https://github.com/knative/eventing/pull/5404/commits/9536f9560d55ecd8cbb5543c86cdc5af1ca48094 I've added some clarifications between normative and non-normative. I think this document should not specify how to choose between normative and non-normative, because IMO depends on the feature itself and the discussions around it. I've just clarified that we cover both. How does it look like?